src/components/add-journey/JourneySearchResults.tsx

```diff
@@ -13,11 +13,41 @@ const generateJourneyKey = (journey: Journey) => {
   return `${journey.from.departureTimestamp}${journey.from.departure}${journey.to.arrivalTimestamp}${journey.to.arrival}`;
 };
 
+const sortJourneys = (a: Journey, b: Journey) => {
+  return a.from.departureTimestamp - b.from.departureTimestamp;
+};
+
+const unionJourneys = (journeys: Journey[], newJourneys: Journey[]) => {
+  const concatJourneys = journeys.concat(newJourneys);
+  const filteredJourneys = concatJourneys.filter(
+    (item, idx) =>
+      concatJourneys.map((journey) => journey.from.departureTimestamp).indexOf(item.from.departureTimestamp) === idx
+  );
+  return filteredJourneys.sort(sortJourneys);
+};
+
 const ResultDisplay: React.FC = () => {
   const journeys = useJourneySearchStore((state) => state.journeys);
+  const setJourneys = useJourneySearchStore((state) => state.setJourneys);
   const departureTime = useJourneySearchStore((state) => state.departureTime);
-  const setDepartureTime = useJourneySearchStore((state) => state.setDepartureTime);
-  const { isFetching, refetch: getJourneys } = useGetJourneys();
+  const departureStation = useJourneySearchStore((state) => state.departureStation);
+  const arrivalStation = useJourneySearchStore((state) => state.arrivalStation);
+
+  const { isLoading, mutateAsync } = useGetJourneys();
+
+  const departureTimeDate = new Date(zonedTimeToUtc(departureTime, 'Europe/Zurich'));
+  const departureTimestamps = (journeys || [])
+    .map((journey) => journey.from.departureTimestamp)
+    .concat(departureTimeDate.getTime() / 1000);
+
+  const minTimestamp = departureTimestamps.length > 0 ? Math.min(...departureTimestamps) : undefined;
+  const earliestDepartureTime = minTimestamp
+    ? formatInTimeZone(new Date(minTimestamp * 1000), 'Europe/Zurich', "yyyy-MM-dd'T'HH:mm")
+    : departureTime;
+  const maxTimestamp = departureTimestamps.length > 0 ? Math.max(...departureTimestamps) : undefined;
+  const latestDepartureTime = maxTimestamp
+    ? formatInTimeZone(new Date(maxTimestamp * 1000), 'Europe/Zurich', "yyyy-MM-dd'T'HH:mm")
+    : departureTime;
```

Comment:
had to do this rather hacky set of functions to determine the earliest / latest timestamp with timezone conversions. There's probably an easier way to do this, like storing data in UTC or something, but worked with what I had